sound/soc/sof/ipc4-topology.c

```diff
@@ -1362,7 +1362,8 @@ static int snd_sof_get_nhlt_endpoint_data(struct snd_sof_dev *sdev, struct snd_s
 	switch (linktype) {
 	case SOF_DAI_INTEL_DMIC:
 		nhlt_type = NHLT_LINK_DMIC;
-		bit_depth = params_width(params);
+		/* always use 32-bit for DMIC bit depth */
+		bit_depth = 32;
```

Comment:
Humm, didn't we have an issue recently where the NHLT blobs extracted from ACPI tables used 24 bits?

Reply:
but wasn't that only the valid bits? And the patch made the change to ignore the valid bits for DMIC

Reply:
What is wrong with 16->16 bit conversion when the NHLT blob supports it? Unlikely but what will happen if the blob contains 16bit DMIC support?
Are you saying that all existing NHLT blob oly supports 32/24 bit and never had 16 bit configuration?

Reply:
What if there is no 32 bit blob in the BIOS or topology? I think the capture DAI copier should generally (not just DMIC) try to look for highest quality enabled format for the copier, and proceed to lower if not found. First 32, then 24, last 16. Is such logic possible if several blobs are available? Used host PCM format should not be the blob selection criteria.

In audio pre-processing for voice it's a good idea to capture from DAI with a high precision and then after all the DRC and noise reduction processing narrow the word length in host copier, e.g. to usual 16 bits if prefer to minimize the audio data bandwidth.

Reply:
@singalsu yes its possible. I've updated the PR to use the 32-bit blob if available first and then fallback to the other formats otherwise.

Reply:
> What is wrong with 16->16 bit conversion when the NHLT blob supports it? Unlikely but what will happen if the blob contains 16bit DMIC support? Are you saying that all existing NHLT blob oly supports 32/24 bit and never had 16 bit configuration?

yes, with topology2 we've never had a 16-bit blob. But I cant say the same thing about the BIOS NHLT

Reply:
@ranj063, the other option would be to have proper tplg files to begin with?
I mean, if we want to allow 16-bit, the tplg should contain 16-bit blob.

Reply:
@ujfalusi the point that @singalsu is trying to make is that when a 32-bit blob is available, we always use that and only fallback to 16-bit when it isnt.